pkg/agent/bakerapi.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package agent
+
+import (
+	"context"
+
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+)
+
+type AgentBaker interface {
+	GetNodeBootstrapping(ctx context.Context, config *datamodel.NodeBootstrappingConfiguration) (*datamodel.NodeBootstrapping, error)
+}
+
+func NewAgentBaker() (AgentBaker, error) {
+	return &agentBakerImpl{}, nil
+}
+
+type agentBakerImpl struct{}
+
+func (agentBaker *agentBakerImpl) GetNodeBootstrapping(ctx context.Context,
+	config *datamodel.NodeBootstrappingConfiguration) (*datamodel.NodeBootstrapping, error) {
+	templateGenerator := InitializeTemplateGenerator()
+	nodeBootstrapping := &datamodel.NodeBootstrapping{
+		CustomData: templateGenerator.GetNodeBootstrappingPayload(config),
+		CSE:        templateGenerator.GetNodeBootstrappingCmd(config),
+	}
+
+	if osImageConfigMap, hasCloud := datamodel.AzureCloudToOSImageMap[config.CloudSpecConfig.CloudName]; hasCloud {
+		if osImageConfig, hasImage := osImageConfigMap[config.AgentPoolProfile.Distro]; hasImage {
+			nodeBootstrapping.OSImageConfig = osImageConfig
+		}
+	}
+
+	return nodeBootstrapping, nil
```

Comment:
do we need to return an error if the input distro doesn't have a corresponding AzureCloudOSImage?

Reply:
actually do we need to move SIG configs here as well? Will that be done in a separate PR?

Reply:
I changed the code to return an error when cloud is not found.

Not returning an error if image is not found yet because for some cases the image is only found in SIG.

I'll prepare a new PR to move over the SIG settings. When that is done, if the image is not found in both places, I'll make the function return an error.